src/hooks/useWeather.jsx

```diff
@@ -0,0 +1,36 @@
+import axios from "axios";
+import {useEffect, useState} from "react";
+import {API_KEY, API_URL} from "../constants/const.ts";
+
+export function useWeather(lat, lon) {
+    const [weatherData, setWeatherData] = useState(null);
+    const [icon, setIcon] = useState(null);
```

Comment:
Rarely use null, 
if you are waiting for an object, specify `{}`
if some string then `''`
number =` 0 (or other initial)`
etc.